testing/CMakeLists.txt

```diff
@@ -101,29 +114,88 @@ add_library(libbmc  SHARED
 
   
 add_library(fpgad-static
-    ${OPAE_SDK_SOURCE}/tools/base/fpgad/ap6.c
-    ${OPAE_SDK_SOURCE}/tools/base/fpgad/ap_event.c
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/command_line.c
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/config_file.c
     ${OPAE_SDK_SOURCE}/tools/base/fpgad/daemonize.c
-    ${OPAE_SDK_SOURCE}/tools/base/fpgad/errtable.c
-    ${OPAE_SDK_SOURCE}/tools/base/fpgad/evt.c
-    ${OPAE_SDK_SOURCE}/tools/base/fpgad/log.c
-    ${OPAE_SDK_SOURCE}/tools/base/fpgad/srv.c
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/event_dispatcher_thread.c
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/events_api_thread.c
     ${OPAE_SDK_SOURCE}/tools/base/fpgad/fpgad.c
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/monitored_device.c
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/monitor_thread.c
     )
 
 target_compile_definitions(fpgad-static PRIVATE main=fpgad_main)
 
 target_include_directories(fpgad-static PUBLIC
   $<BUILD_INTERFACE:${OPAE_INCLUDE_DIR}>
   $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
+  PRIVATE ${OPAE_SDK_SOURCE}/libbitstream
   $<INSTALL_INTERFACE:include>)
 
 target_link_libraries(fpgad-static
     safestr
+    bitstream-static
     ${libjson-c_LIBRARIES}
     ${libuuid_LIBRARIES}
     )
 
+add_library(fpgad-api-static
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/api/logging.c
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/api/opae_events_api.c
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/api/device_monitoring.c
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/api/sysfs.c
+    ) 
+
+target_include_directories(fpgad-api-static PUBLIC
+  $<BUILD_INTERFACE:${OPAE_INCLUDE_DIR}>
+  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+  PRIVATE ${OPAE_SDK_SOURCE}/libopae
+  PRIVATE ${OPAE_SDK_SOURCE}/libbitstream
+  PRIVATE ${OPAE_SDK_SOURCE}/tools/base
+  PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+
+target_link_libraries(fpgad-api-static
+    safestr
+    ${CMAKE_THREAD_LIBS_INIT}
+    ${libjson-c_LIBRARIES}
+    )
+
+add_library(fpgad-xfpga-static
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/plugins/fpgad-xfpga/fpgad-xfpga.c
+    ) 
+
+target_include_directories(fpgad-xfpga-static PUBLIC
+  $<BUILD_INTERFACE:${OPAE_INCLUDE_DIR}>
+  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+  PRIVATE ${OPAE_SDK_SOURCE}/libopae
+  PRIVATE ${OPAE_SDK_SOURCE}/libbitstream
+  PRIVATE ${OPAE_SDK_SOURCE}/tools/base
+  )
+
+target_link_libraries(fpgad-xfpga-static
+    fpgad-api-static
+    ${CMAKE_THREAD_LIBS_INIT}
+    )
+
+add_library(fpgad-vc-static
+    ${OPAE_SDK_SOURCE}/tools/base/fpgad/plugins/fpgad-vc/fpgad-vc.c
+    ) 
+
+target_include_directories(fpgad-vc-static PUBLIC
+  $<BUILD_INTERFACE:${OPAE_INCLUDE_DIR}>
+  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+  PRIVATE ${OPAE_SDK_SOURCE}/libopae
+  PRIVATE ${OPAE_SDK_SOURCE}/tools/base
+  )
```

Comment:
The fpgad-vc-static is unable to build, missing libbitstream.

Reply:
Oh, it's missing the header.  Thanks, will fix.